### PR TITLE
[bullshark] fix sub-dag leader round calculation

### PR DESCRIPTION
**Pull request by mwtian**
Currently leaders / anchors for sub-dags are selected from odd rounds, because the first sub-dag leader is selected at leader's `round - 1` instead of `round - 2`. Fix the range expression to generate sub-dag leader rounds at even rounds.

Will try to see if a unit test can be added.

### CODE REVIEW

File: narwhal/consensus/src/utils.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ where
 {
     let mut to_commit = vec![leader.clone()];
     let mut leader = leader;
-    for r in (state.last_committed_round + 2..leader.round())
+    for r in (state.last_committed_round + 2..=leader.round() - 2)
```
Comment (asonnino) [inline narwhal/consensus/src/utils.rs:22]: Why this change? Isn't it the same thing?

Comment (mwtian) [reply narwhal/consensus/src/utils.rs]: @asonnino should I close this, or is this a no-op?

Comment (asonnino) [reply narwhal/consensus/src/utils.rs]: oh I missed your comment, checking now

**Review by asonnino — APPROVED** [review]

File: narwhal/consensus/src/utils.rs (hunk 1 above)
Comment (asonnino) [reply narwhal/consensus/src/utils.rs]: That seems a bug (the `rev` got me confused), a dangerous one

Comment (mwtian) [reply narwhal/consensus/src/utils.rs]: I see. Will try to add more tests to catch issues like this.

File: narwhal/consensus/src/utils.rs [hunk 2]
```diff
@@ -19,7 +19,8 @@ where
 {
     let mut to_commit = vec![leader.clone()];
     let mut leader = leader;
-    for r in (state.last_committed_round + 2..leader.round())
+    assert_eq!(leader.round() % 2, 0);
+    for r in (state.last_committed_round + 2..=leader.round() - 2)
```
Comment (akichidis) [inline narwhal/consensus/src/utils.rs:23]: The fact that no test broke because of this it tells me that we are definitely missing an important test case here
